app/models/spree/gateway/flow_io.rb

```diff
@@ -56,18 +58,28 @@ def purchase(_amount, _payment_method, options = {})
         end
       end
 
-      def refund(_money, _authorization_key, options = {})
-        order = load_order options
-        order.cc_refund
+      def refund(payment, amount, _options = {})
+        order = payment.order
+        refund_form =
+          Io::Flow::V0::Models::RefundForm.new(order_number: order.number, amount: amount, currency: order.currency)
+        response = FlowcommerceSpree.client.refunds.post(FlowcommerceSpree::ORGANIZATION, refund_form)
+        response_status = response.status.value
+        if response_status == REFUND_SUCCESS
+          add_refund_to_order(response, order)
+          ActiveMerchant::Billing::Response.new(true, REFUND_SUCCESS, {}, {})
+        else
+          msg = "Partial refund fail. Details: #{response_status}"
+          ActiveMerchant::Billing::Response.new(false, msg, {}, {})
+        end
+      rescue StandardError => e
+        ActiveMerchant::Billing::Response.new(false, e, {}, {})
```

Comment:
Is the message in the `ActiveMerchant::Billing::Response` used in the front end? In case it is, can we use a more descriptive sentence?

Reply:
Nope, this is for Spree::Admin